First draft documenting motivation (#23)
ietf-rats/draft-deshpande-multi-verifier: draft-deshpande-rats-multi-verifier.md

# Introduction

> A Verifier plays a central role in any Remote Attestation System. A Verifier appraises the Attester and produces Attestation Results, which is essentially a verdict of attestation. The results are consumed by the Relying Party to conclude the trustworthiness of the Attester, prior to making any critical decisions about the Attester, such as admitting to network or releasing confidential resources to it.

Reviewer: ```suggestion
A Verifier plays a central role in any Remote Attestation System. A Verifier appraises the Attester and produces Attestation Results, which are essentially a verdict of attestation. The results are consumed by the Relying Party to conclude the trustworthiness of the Attester, before making any critical decisions about the Attester, such as admitting it to the network or releasing confidential resources to it.
```

Attesters can come in wide varieties of shape and form. For example Attesters can be endpoints (edge or IoT devices) or complex machines in the cloud. Composite Attester {{sec-glossary}}, generate Evidence that consists of multiple parts. For example, in data center servers, it is not uncommon for separate attesting environments (AE) to serve a subsection of the entire machine. One AE might measure and attest to what was booted on the main CPU, while another AE might measure and attest to what was booted machine's GPU. Throughout this document we use the term Component Attester {{sec-glossary}} to address the sub-entity or an individual layer which produces its own Evidence in a Composite Attester system.

> In a Composite Attester{{sec-glossary}} system it may not be possible that a single Verifier may possess all the capabilities or the information required to conduct the complete appraisal of the Attester. Please refer to {{sec-need-multiverifier}}, for motivation of this document. Multiple Verifiers need to collaborate to conclude the appraisal and produce the Attestation Results.

Reviewer: ```suggestion
In a Composite Attester system, it may not be possible for a single Verifier to possess all the capabilities or information required to conduct a complete appraisal of the Attester. Please refer to {{sec-need-multiverifier}} for motivation of this document. Multiple Verifiers need to collaborate to reach a conclusion on the appraisal and produce the Attestation Results.
```

This document describes various topological patterns of multiple Verifiers that work in a coordinated manner to conduct appraisal of a Composite Attester to produce an Attestation Results.

# Need for Multiple Verifiers
{: #sec-need-multiverifier }

> In order to conduct the role of attestation verification, a Verifier needs:

Reviewer 2: ```suggestion
In order to conduct the task of evidence appraisal, a Verifier needs:
```

> 1. Reference Values from trusted supply chain of the Attester
> 2. Endorsements from trusted supply chain of the Attester
> 3. Appraisal policy for Evidence, which is under the control of Verifier owner

Reviewer 2: ```suggestion
1. Reference Values from trusted supply chain actors producing, aggregating, selling, or administrating Attesters (Reference Value Providers)
2. Endorsements from trusted supply chain actors producing, certifying, or compliance checking Attesters (Endorsers)
3. Appraisal Policy for Evidence, which is under the control of Verifier Owner
```

> 

Reviewer 3: 4. Certificates for Attesting Environments.

> The above inputs are linked to the shape and form of the Attester, i.e the type of claims and the policies of Verification.

Reviewer 2: ```suggestion
The Verifier inputs listed above are linked to the shape and form of the Attesters, i.e., the type of claims and the policies of appraisal of Evidence.
```

> For composite Attesters, there is a heterogeneity of Evidence formats, each with a specific purpose, for example a CPU and a GPU.

Reviewer 2: ```suggestion
Typically, composite Attesters come with a varying degree of heterogeneity of Evidence formats, depending on the type of Attesting Environments that come with Attester components, for example, CPU variants or GPU/FPGA variants.
```

Reviewer: I think we have "Component Attester” in the glossary, rithg?  If not, we shldou :-)

> The composite Attester composition can change based on market dynamics and availability over time (e.g., a set of racks in a

Reviewer 2: ```suggestion
An Attester's composition can over time change based on market dynamics and availability (e.g., a set of racks in a
```

Reviewer: Instead of “market dynamics” I think “deployment updates” is preferable.

Author: No I think, market dynamics is more suited that goes to the root why it changes, instead `deployment updates` which sort of reflects the action..

> 1. An Attester's composition can change over time based on market dynamics and availability (e.g., a set of racks in a data center gets thousands of new FPGAs).

Author: ```suggestion
1. An Attester's composition can change over time based on market requirements and availability (e.g., a set of racks in a data center gets thousands of new FPGAs).
```

> data center gets thousands of new FPGAs). It is highly unlikely that there is always one appropriate Verifier that satisfies all the requirements that a complex and changing composite Attesters impose. It may not be economically viable to build and maintain such degree of complexity, in a single Verifier, hence there is a need for a Multi Verifier System.

Reviewer: it looks like  we don’t do NSNL here.  What’s the reason?

In addition to above, there are several other factors that can lead to multi Verifier system, few are listed below:
1. A part of a single component Verification policy may be internal to a Verifier Owner and it may not choose to reveal it to a monolithic Verifier.
2. An Endorser may not wish to reveal its Reference Values or their lifecycle to a monolithic Verifier, hence may choose to keep its verification seperate.

> 2. There may not be a single actor in the ecosystem, that can stand up and take ownership of Verifying every component Attester due to lack of knowledge, complexity, regulations or associated cost.

Author: ```suggestion
3. There may not be a single actor in the ecosystem, that can stand up and take ownership of Verifying every component Attester due to lack of knowledge, complexity, regulations or associated cost.
```

> 3. Verifier services provided by component manufactures as well as Verifiers provided by the integrators, as well as Verifiers under local authority (close to the Attester) are the mix today and rarely it is just one of them.

Author: ```suggestion
4. Verifier services provided by component manufactures as well as Verifiers provided by the integrators, as well as Verifiers under local authority (close to the Attester) are the mix today and rarely it is just one of them.
```

> 4. Verifier services provided by component manufactures as well as Verifiers provided by the integrators, as well as Verifiers under local authority (close to the Attester) are the mix today and rarely it is just one of them.

Reviewer: Lines 88-92 *may* be ok in terms of content (although I am not entirely sure about 2), but they need some more work on the exposition to improve clarity.

Besides, I don’t understand why they are grouped separately.  They seem like core arguments to me, i.e., not hierarchically inferior to the “market dynamics” argument above.

In general, I believe it’d be better to restructure the section in a cleaner fashion, e.g., with a bullet list where each separate point is clearly and concisely articulated.

In a Composite Attester system, it may not be possible for a single Verifier to possess all the capabilities or information required to conduct a complete appraisal of the Attester. Please refer to {{sec-need-multiverifier}} for motivation of this document. Multiple Verifiers need to collaborate to reach a conclusion on the appraisal and produce the Attestation Results.

This document describes various topological patterns of multiple Verifiers that work in a coordinated manner to conduct appraisal of a Composite Attester to produce an Attestation Results.

# Need for Multiple Verifiers
{: #sec-need-multiverifier }
To conduct the task of Evidence appraisal, a Verifier requires:
1. Reference Values from trusted supply chain actors producing, aggregating, or administering Attesters (Reference Value Providers)
2. Endorsements from trusted supply chain actors producing, certifying, or compliance checking Attesters (Endorsers)
3. Appraisal Policy for Evidence, which is under the control of the Verifier Owner

> The Verifier inputs listed above are linked to the shape and form of the Attesters, i.e., the type of claims and the policies of appraisal of Evidence.
> Typically, Composite Attesters {{sec-glossary}} come with a varying degree of heterogeneity of Evidence formats, depending on the type of Attesting Environments that come with each Component Attester {{sec-glossary}}, for example, CPU variants or GPU/FPGA variants. When conducting appraisal of Evidence from a Composite Attester, following challenges remain.
> 

Reviewer: ```suggestion
The Verifier inputs listed above are linked to the shape of the Attesters.
Typically, Composite Attesters come with a varying degree of heterogeneity of Evidence formats, depending on the type of Attesting Environments that come with each Component Attester, for example, CPU variants or GPU/FPGA variants. When conducting Evidence appraisal for a Composite Attester, the following challenges remain:

```

> 1. An Attester's composition can change over time based on market dynamics and availability (e.g., a set of racks in a
> data center gets thousands of new FPGAs).
> It is highly unlikely that there is always one appropriate Verifier that satisfies all the requirements that a complex and changing composite Attesters impose.
> It may not be economically viable to build and maintain such degree of complexity, in a single Verifier, hence there is a need for a Multi Verifier System.
> 2. A part of a single component Verification policy may be internal to a Verifier Owner and it may not choose to reveal it to a monolithic Verifier.
> 3. An Endorser may not wish to reveal its Reference Values or their lifecycle to a monolithic Verifier, hence may choose to keep its verification seperate.
> 4. There may not be a single actor in the ecosystem, that can stand up and take ownership of Verifying every component Attester due to lack of knowledge, complexity, regulations or associated cost.
> 5. Verifier services provided by component manufactures as well as Verifiers provided by the integrators, as well as Verifiers under local authority (close to the Attester) are the mix today and rarely it is just one of them.

Reviewer: ```suggestion
1. An Attester's composition can change over time based on market dynamics and availability (e.g., a set of racks in a data center gets thousands of new FPGAs).
It is highly unlikely that there is always one appropriate Verifier that satisfies all the requirements that a complex and changing Composite Attesters imposes.
It may not be economically viable to build and maintain such a degree of complexity in a single Verifier.
2. A Verifier Owner may have an Appraisal Policy for Evidence of a Component Attester that is internal to them and which they may choose not to reveal to a “monolithic" Verifier.
3. A Reference Values Provider may not wish to reveal its Reference Values or their lifecycle to a monolithic Verifier.
4. There may not be a single actor in the ecosystem that can stand up and take ownership of verifying every Component Attester due to a lack of knowledge, complexity, regulations or associated cost.
5. The mix today is a combination of Verifier services provided by component manufacturers, Verifiers provided by integrators, and Verifiers under local authority (i.e., close to the attester). Rarely is it just one of these.
```
